### PR TITLE
Remove `as frame_system::DefaultConfig` from the required syntax in `derive_impl`

### PR DESCRIPTION
**Pull request by gupnik**
Step in https://github.com/paritytech/polkadot-sdk/issues/171

This PR removes the need to specify `as [disambiguation_path]` for cases where the trait definition resides within the same scope as default impl path. 

For example, in the following macro invocation
```rust
#[derive_impl(frame_system::config_preludes::TestDefaultConfig as frame_system::DefaultConfig)]
impl frame_system::Config for Runtime {
   ...
}
```
the trait `DefaultConfig` lies within the `frame_system` scope and `TestDefaultConfig` impls the `DefaultConfig` trait. Using this information, we can compute the disambiguation path internally, thus removing the need of an explicit specification.

In cases where the trait lies outside this scope, we would still need to specify it explicitly, but this should take care of most (if not all) uses of `derive_impl` within FRAME's context.

### CODE REVIEW

**Review by liamaharon — APPROVED** [review]
Nice one! Can we do it for all cases in polkadot-sdk?

**Comment by gupnik** [comment]
> Nice one! Can we do it for all cases in polkadot-sdk?

Yes, will do it in a separate PR to make the review easier.

File: substrate/frame/support/procedural/src/derive_impl.rs [hunk 1]
```diff
@@ -172,6 +172,31 @@ fn combine_impls(
 	final_impl
 }
 
+fn compute_disambiguation_path(
```
Comment (ggwpez) [inline substrate/frame/support/procedural/src/derive_impl.rs:175]: Some internal docs please.

**Review by ggwpez — APPROVED** [review]
💯

File: substrate/frame/support/procedural/src/derive_impl.rs (hunk 1 above)
Comment (gupnik) [reply substrate/frame/support/procedural/src/derive_impl.rs:175]: Added in the code but will add here as well

**Pull request merged**